server/src/main/java/org/opensearch/common/cache/tier/EhCacheDiskCachingTier.java

```diff
@@ -106,8 +127,17 @@ private EhCacheDiskCachingTier(Builder<K, V> builder) {
         this.DISK_WRITE_CONCURRENCY = Setting.intSetting(builder.settingPrefix + ".tiered.disk.ehcache.concurrency", 2, 1, 3);
         // Default value is 16 within Ehcache.
         this.DISK_SEGMENTS = Setting.intSetting(builder.settingPrefix + ".ehcache.disk.segments", 16, 1, 32);
-        this.cacheManager = buildCacheManager();
+
+        // In test cases, there might be leftover cache managers and caches hanging around, from nodes created in the test case setup
+        // Destroy them before recreating them
+        close();
+        cacheManager = buildCacheManager();
         this.cache = buildCache(Duration.ofMillis(expireAfterAccess.getMillis()), builder);
+
+        // IndicesRequestCache gets 1%, of which we allocate 5% to the keystore = 0.05%
+        // TODO: how do we change this automatically based on INDICES_CACHE_QUERY_SIZE setting?
```

Comment:
U can pass IRC size into this class while initialization? Or a better approach be passing the keystore size during initialization?